### ISSUE TITLE
Mass Mint: Not working as it should?

### ISSUE THREAD

**Issue by JustLuuuu** [kodadot/nft-gallery#7184]
Two artists were trying mass-mint (on production) on Asset Hub Kusama:
1. https://kodadot.xyz/ahk/collection/119 (tg: @MxMyA)
2. https://kodadot.xyz/ahk/collection/123 (https://twitter.com/Artistken1)

Both of them did what the tutorial said. They uploaded zip files with the NFTs.

## First artist:
Uploaded zip file with pics, he didn‘t upload a text or json file, but modified the data one by one in the third section (table) then submitted.
The collection of the first artist now looks like this:
<img width="1189" alt="Screenshot 2023-09-06 at 12 35 35" src="https://github.com/kodadot/nft-gallery/assets/104380776/dacfbd3a-a624-4c9b-8c6c-5535597d6248">
The NFTs were supposed to look like this:
![2023-09-06 12 33 28](https://github.com/kodadot/nft-gallery/assets/104380776/e4958eb5-d113-48ff-bc11-72e48f6e5a31)

## Second artist wrote me this
Quite a few of the thumbnails have not rendered yet. Not sure why. All the images were in a .zip.

These are new artists for Berlin exhibition, the art needs to be minted properly by Friday.

**Comment by roiLeo** [comment]
Share zip file content if you can so we can reproduce the bug.

**Comment by JustLuuuu** [comment]
> Share zip file content if you can so we can reproduce the bug.

First artist: zip file 
[first_thirty.zip](https://github.com/kodadot/nft-gallery/files/12539807/first_thirty.zip) Uploaded zip file with pics, he didn‘t upload a text or json file, but modified the data one by one in the third section (table) then submitted.

**Comment by JustLuuuu** [comment]
> Share zip file content if you can so we can reproduce the bug.

Second artist:
- zip file: https://drive.google.com/file/d/1yqeMAkGVLCKAwKdMbZJXZjrmx3Q-6g8C/view
- text file: https://drive.google.com/file/d/1n-rMgLOBUWVv2pi6nEvFZBamBGsbxazq/view
- "I had to input some of the image info manually but they all showed good to go. Everything seemed normal until I check the collection."
- the original files: https://drive.google.com/drive/folders/1FJH9OYezoMhV21Xzb9hC1LlWCpUJwc-W

**Comment by daiagi** [comment]
all the items with wrong image were minted with empty metadata 
but then it seems that in the absence of metadata for the nft, the indexer falls back to the collection metadata @vikiival 

this nft for example https://kodadot.xyz/ahk/gallery/119-2
the image and description are those of the collection

**Comment by kodabot** [comment]
ASSIGNED - @daiagi 🔒 LOCKED -> Saturday, September 9th 2023, 17:17:39 UTC -> 36 hours

**Comment by daiagi** [comment]
there is our issue:
500 on POST to https://nft-storage.kodadot.workers.dev/pinFile 


![image](https://github.com/kodadot/nft-gallery/assets/22791238/9ef7b173-bae8-4604-ab1b-74f7eb6b4cc0)

![image](https://github.com/kodadot/nft-gallery/assets/22791238/c29c8bef-c670-4c05-8179-3ea9b1af7df7)


a. @vikiival  please advise
b. prevent empty metadata nft's from getting minted, show error to user

**Comment by JustLuuuu** [comment]
Another [artist](https://twitter.com/HookerNFT) used mass mint:
https://kodadot.xyz/ahk/u/HhwNst3i5aG8T5am3hVRT7fX9MRTUCcE2BaujXoxCmdpQVe

His collection is looking like this:
<img width="1409" alt="Screenshot 2023-09-08 at 09 22 06" src="https://github.com/kodadot/nft-gallery/assets/104380776/7acd8333-f842-4e11-bca0-07f02bb98482">

**Comment by vikiival** [comment]
> indexer falls back to the collection metadata @vikiival

Yes that is expected. People minted a lot of nfts without metadata to save on deposits .)

**Comment by daiagi** [comment]
more info:
@vikiival 

getKey requestes are randonly failing (500)
the extension metadata is updated
![image](https://github.com/kodadot/nft-gallery/assets/22791238/092f7cd2-4023-4089-93cd-2c7b9a5606d7)



pinFile requests are failing:

![image](https://github.com/kodadot/nft-gallery/assets/22791238/cc38cf40-e031-4f52-9f29-5ebb71fd4596)

**Comment by JustLuuuu** [comment]
> > indexer falls back to the collection metadata @vikiival
> 
> Yes that is expected. People minted a lot of nfts without metadata to save on deposits .)

I think they were not planning to save on the deposit. We were covering all their minting fees. Maybe the step was not clear for them. Anyways, if it a problematic to load nfts whiteout metadata, the mint should never have go through. There should be some notification explaining them what is missing. What now?
- Is it possible to recover their files?
- Do they need to mint from the scratch?

**Comment by vikiival** [comment]
> Do they need to mint from the scratch?

No they don't, there is one trick for that as we do not have UI

**Comment by daiagi** [comment]
> the mint should never have go through. There should be some notification explaining them what is missing. What now?


That's correct
or perhaps even prevent all mints if some NFTs are missing metadata as it is an indication of a problem in the storage worker

i.e. is it better maybe to show a message along the lines of "we are experiencing an internal problem and cannot mint at the moment?"

Maybe that better than minting partiality

@exezbcz  @prury  wdyt?

**Comment by vikiival** [comment]
> getKey requestes are randonly failing (500)

This call can be skipped tbh

> pinFile requests are failing

What do I think is that mass mint is ddosing the service and cloudlflare/nftstorage will rate limit you

Third thing

<img width="767" alt="Screenshot 2023-09-08 at 09 04 09" src="https://github.com/kodadot/nft-gallery/assets/22471030/b3fe2425-209c-495d-a85c-9b8a4320bc91">

Looks like you are not constructing correct payload for the storage

if you are pinning JSON please use 

```
POST /pinJson
```

with body of the json


If it's failing on image may check what's the size. But 30MB+ images are fine (have tested).

**Comment by mxmya** [comment]
Hi,
I would like to share with you some details about my mint experience (mentioned above)

First step was to upload the ZIP file containing the images.
The second step was not possible, since I didn't know the structure of the required TXT or JSON files.
The preview table was activated in the third step and I changed all the inputs including the names, description and prices, manually and one by one throughout the table. A preview of each image was visible.
First when the table was successully filled out, the mint button appeared (indicating to me that the metadata is now full and ready)
After pressing mint, the page kept hanging for hours with the swirling square tails.
I opened the previously created collection in a new window. The NFTs were minted with the collection logo on each.
